homeassistant/components/modbus/base_platform.py

```diff
@@ -232,14 +236,20 @@ def unpack_structure_result(self, registers: list[int]) -> str | None:
                 if isinstance(v_temp, int) and self._precision == 0:
                     v_result.append(str(v_temp))
                 elif v_temp is None:
-                    v_result.append("")  # pragma: no cover
+                    v_result.append("0")
                 elif v_temp != v_temp:  # noqa: PLR0124
                     # NaN float detection replace with None
-                    v_result.append("nan")  # pragma: no cover
+                    v_result.append("nan")
```

Comment:
Don't we have the same problem here?

Reply:
yup, I did change that too, but seems I forgot to press "save".

Done.